### PR TITLE
[Feat] Add sponsors' logo in index page

### PR DESCRIPTION
**Pull request by josix**
* include _inlcudes/sponsors.html in index.html
* Add new sponsors.html
* Update popup card border, close icon style, link-to-website btn style
* Add grid style to sponsors logos

(Thanks for sending a pull request! Please fill in the following content to let us know better about this change.)

## Types of changes
Please put an `x` in the box that applies

- [ ] **Bugfix**
- [x] **New feature**
- [ ] **Refactoring**
- [ ] **Breaking change** (any change that would cause existing functionality to not work as expected)
- [ ] **Documentation Update**
- [ ] **Other (please describe)**

## Description
**Describe what the change is**
You don't need to explain HOW you change the code. Your code, including comments in the code should explicitly
self-describe enough about how you change the code.

* Add new sponsors.html with grid layout style
* Update popup card border, close icon style, link-to-website btn style

## Steps to Test This Pull Request
1. Running command `python3 manange.py runserver`
2. Open your preferred browser and go to `localhost:8000`
3. Scroll down in this page
4. Click any sponsor logo to see the popup-window

## Expected behavior
The view should looks like following screenshots
image
image

The Popup-window should looks like the following screenshot, and all buttons should work normally
image
image

## Related Issue
If applicable, refernce to the issue related to this pull request.
#814 

## More Information
**Screenshots**

**Additional context**
Add any other context about the problem here. You may also want to refer
to [how to wirte the perfect pull request](https://github.blog/2015-01-21-how-to-write-the-perfect-pull-request/)

### CODE REVIEW

File: src/static/pycontw-2020/styles/_index-partners.scss [hunk 1]
```diff
@@ -2,113 +2,60 @@
...
-					vertical-align: middle;
-					font-size: 175%;
-
-					@include on-desktop() {
-						max-width: 100%;
+			
+			column-gap: 40px;
+			row-gap: 40px;
+			margin: 40px;
+			> .partner-container {
```
Comment (yychen) [inline src/static/pycontw-2020/styles/_index-partners.scss:19]: 這邊我建議把 flexbox 內容物置中，例如加上
```scss
justify-content: center;
align-items: center;
```
原因是因為，如果 icon 是正方形的，似乎就會有一些問題。

但是下面的 span 就會變成也是置中，所以提議把 span 變成 absolute，但上面的 `.partner-container` 就要加上 relative

```scss
position: relative;
```

(上面的 style 都是給 `.partner-container` 的)

File: src/static/pycontw-2020/styles/_index-partners.scss [hunk 2]
```diff
@@ -2,113 +2,60 @@
...
+				}
+				> span.silver {
+					color: #262727;
+				}
+				> span.bronze {
+					color: #502526;
+				}
+				> span.special {
+					color: #502526;
+				}
```
Comment (yychen) [inline src/static/pycontw-2020/styles/_index-partners.scss:40]: 這邊可能要有個通則，例如
```scss
> span {
    position: absolute;
    left: 0.5em;
    top: 0.5em;
}
```
(0.5em 是我隨便抓的距離)

File: src/static/pycontw-2020/styles/_index-partners.scss [hunk 3]
```diff
@@ -2,113 +2,60 @@
...
+					img, > span {
+						margin: 40px 0px;
+						position: relative;
+						z-index: $background-deco-index + 1;
+						min-width: 120px;
+						max-width: 200px;
+						height: 64px;
+						line-height: 64px;
+						vertical-align: middle;
+						font-size: 175%;
```
Comment (yychen) [inline src/static/pycontw-2020/styles/_index-partners.scss:51]: 這一塊針對圖片的部分可能弄成
```scss
z-index: $background-deco-index + 1;
max-width: 180px;
max-height: 180px;
```
就好了。

這樣子的話橫的或直的圖片應該都可以 (不同比例)，剛剛有稍微測過一下。
可以到 https://worldvectorlogo.com/ 隨便下一些 logo 測測看。(注意比例，圖片檔案比例本身是橫的方的都測測看，我剛剛就下載到一個 logo 本身是橫的但是他的圖片卻是方形的，他把上下都留白了，這種就要算是方形的圖片)
不過文字的部分我就還沒測過這樣子的規則。

BTW... 通常 logo 檔案都會比這個格子來得大。所以應該不用設定 min-width (?)

File: src/static/pycontw-2020/styles/_index-partners.scss [hunk 4]
```diff
@@ -2,113 +2,60 @@
 
 .index-partners {
```
Comment (yychen) [inline src/static/pycontw-2020/styles/_index-partners.scss:3]: 如果手機版寬度小於 380px 的話，`.index-partners` 的 padding 就會讓他沒有置中。如圖片所示
圖片

可能要加個規則看 mobile 的時候要如何置中這個 container。

**Comment by tai271828** [comment]
A quick rebuild to test @josix 's branch with @mattwang44 's useful docker-compose improvement shows expected behavior to me. For example,

Selection_048


Selection_050

**Comment by tai271828** [comment]
@josix  if you don't mind you may enable the button like "allow to modify your branch" so we could co-work on this pull request according to each of @yychen  comments.

**Comment by josix** [comment]
> @josix if you don't mind you may enable the button like "allow to modify your branch" so we could co-work on this pull request according to each of @yychen comments.

@tai271828 Thanks for your advice, I've enabled the button like the following image shows. Not sure if you could push codes to this PR.
image

File: src/static/pycontw-2020/styles/_index-partners.scss (hunk 1 above)
Comment (josix) [reply src/static/pycontw-2020/styles/_index-partners.scss:20]: 感謝 @yychen 提醒，沒注意到的 logo size 問題，這邊已修正。

File: src/static/pycontw-2020/styles/_index-partners.scss (hunk 2 above)
Comment (josix) [reply src/static/pycontw-2020/styles/_index-partners.scss:49]: 這邊已經修正，謝謝 Tom。

File: src/static/pycontw-2020/styles/_index-partners.scss (hunk 4 above)
Comment (josix) [reply src/static/pycontw-2020/styles/_index-partners.scss:3]: 我將 padding 統一設為 0 看起來 desktop/mobile 都沒有跑版，應該已經解決這個問題了，謝謝 Tom。

File: src/static/pycontw-2020/styles/_index-partners.scss (hunk 3 above)
Comment (josix) [reply src/static/pycontw-2020/styles/_index-partners.scss]: 對的 min-width 應該是留到去年的 code 忘記刪掉了，感謝提醒，目前我已經設定
```css
z-index: $background-deco-index + 1;
max-width: 200px;
max-height: 200px;
```
不同的 logo 我可能還要再測一下，正方形的 Logo 的測試結果如下看起來應該沒有問題
image
image
image

Comment (yychen) [reply src/static/pycontw-2020/styles/_index-partners.scss]: 這個方形的圖片是本身很大張，四周有透明像素，還是它本身大小就是這樣呢? 感覺他跟其他橫式的 logo 排排站它變得很小 orz

Comment (josix) [reply src/static/pycontw-2020/styles/_index-partners.scss]: 這個是 909x909 的正方形圖，無四周透明，應該是之前設定 height 為 64 導致的，修改後目前會是如下圖：
image

Comment (yychen) [reply src/static/pycontw-2020/styles/_index-partners.scss]: 好好!! 這樣好多了 👍

File: src/static/pycontw-2020/styles/_index-partners.scss (hunk 4 above)
Comment (tai271828) [reply src/static/pycontw-2020/styles/_index-partners.scss:3]: width 300 看起來好像還可以

Selection_191

**Comment by kunxianhuang** [comment]
Hello, 
Here are some comments from Sponsorship group.
Sponsors will probability like the background color to be the same with one of the webpage. (pink)
Also, I found some uploaded svg files can't be shown in the page, so I change to jpg files.
Finally, comparing to last year's logo, the size is smaller. We have not hear any complain from sponsor, but do you consider make the size bigger?

Thanks for your hard work for Sponsor's logo.

Kunxian Huang for the Sponsorship group of PyCon TW

**Comment by yychen** [comment]
> 
> 
> Hello,
> Here are some comments from Sponsorship group.
> Sponsors will probability like the background color to be the same with one of the webpage. (pink)
@kunxianhuang 請問你的意思是說不要有白色方框作為底色嗎? 我想這個是設計師的設計的說 orz
或者是說用成這樣有什麼顧慮嗎?

> Also, I found some uploaded svg files can't be shown in the page, so I change to jpg files.
@josix can you take a look?

> Finally, comparing to last year's logo, the size is smaller. We have not hear any complain from sponsor, but do you consider make the size bigger?
好，我這邊再調看看要多大比較適合。感謝 Feedback :D

> 
> Thanks for your hard work for Sponsor's logo.
> 
> Kunxian Huang for the Sponsorship group of PyCon TW

**Comment by josix** [comment]
Okay 我明天來測

**Comment by kunxianhuang** [comment]
> > Hello,
> > Here are some comments from Sponsorship group.
> > Sponsors will probability like the background color to be the same with one of the webpage. (pink)
> 
> @kunxianhuang 請問你的意思是說不要有白色方框作為底色嗎? 我想這個是設計師的設計的說 orz
> 或者是說用成這樣有什麼顧慮嗎?
> 
Yes, 
the white square has some conflict with scheme of the website, 
I look the website, and found only sponsorship use white block.
It is not a good balance of color in whole website, I think keep transparent is good enough.

> > Also, I found some uploaded svg files can't be shown in the page, so I change to jpg files.
> 
> @josix can you take a look?
> 
If you need the svg file to test, I can send it to you.
> > Finally, comparing to last year's logo, the size is smaller. We have not hear any complain from sponsor, but do you consider make the size bigger?
> 
> 好，我這邊再調看看要多大比較適合。感謝 Feedback :D
> 
I see your size is 100 by 100 (w x h), as the sponsor's log usually is 2X1 (wXh).
So, I would like to have 256 X 128 (wXh).
Does it good for your design team, too?

> > Thanks for your hard work for Sponsor's logo.
> > Kunxian Huang for the Sponsorship group of PyCon TW

Thanks for your reply and work.

Kunxian Huang

**Pull request merged**